### PR TITLE
bench: update benchmarks and examples in `math/base/special/ldexp`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   replaces the use of built-in `pow` with `stdlib_base_pow`.
-   fixes indentation in `example.c`.

## Related Issues

> Does this pull request have any related issues?

None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by gunjjoshi** [comment]
This error in examples comes even without applying the changes in this PR. I'll have a look on `ldexp` to find out the problem.

**Comment by gunjjoshi** [comment]
The error message comes as:

```bash
example.c:20:10: fatal error: 'stdlib/math/base/special/ldexp.h' file not found
#include "stdlib/math/base/special/ldexp.h"
```
even though the header files looks correct.

**Comment by Planeshifter** [comment]
@gunjjoshi Looks like `ldexp` is missing from the `manifest.json`' dependencies in the `example` configuration.

File: lib/node_modules/@stdlib/math/base/special/ldexp/examples/c/example.c [hunk 1]
```diff
@@ -34,29 +32,29 @@ static double rand_double() {
...
+	double frac;
+	int32_t exp;
+	double x;
+	double v;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		if ( rand_double() < 0.5 ) {
+			sign = -1.0;
+    		} else {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/ldexp/examples/c/example.c]:
```suggestion
			} else {
```

**Comment by gunjjoshi** [comment]
> @gunjjoshi Looks like `ldexp` is missing from the `manifest.json`' dependencies in the `example` configuration.

@Planeshifter For other packages too, we do not include the package's path in the `example` configuration of `manifest.json`, right? That is included directly from the header file inside the package. Still, I tried after including that, but the condition was the same.

**Comment by Planeshifter** [comment]
@gunjjoshi Yeah, the package itself shouldn't be listed as a dependency under its own configuration. I was mistakenly under the impression that the failure was caused in a different package's example. Not clear to me what's going on right now... Also seeing failures loading different modules in the CI checks here.

**Comment by Planeshifter** [comment]
@gunjjoshi `@stdlib/number/float32/base/to-word` seems wrong as a dependency in `manifest.json`. We need `@stdlib/number/float64/base/to-words`, I think.

**Comment by gunjjoshi** [comment]
@Planeshifter It doesn't seem we have used that package anywhere in `ldexp`. We should remove that.

**Comment by gunjjoshi** [comment]
We have used `float64/base/to_words` in `main.c`, but we listed `float32/base/to_word` in `manifest.json`. Even after rectifying that, the error persists.

**Comment by Planeshifter** [comment]
After looking into this a bit more, I noticed that running the C examples for `@stdlib/math/base/special/pow` is failing as well.

Running the examples with `DEBUG=library-manifest:main`, it looks to me as if we have a cycle here and `@stdlib/utils/library-manifest` may not be handling those yet? 

cc @kgryte

**Comment by kgryte** [comment]
> it looks to me as if we have a cycle here and `@stdlib/utils/library-manifest` may not be handling those yet?

That's correct. We cannot handle cyclic dependencies in C implementations. In this case, I think it would be fine to refactor the example in order to avoid the use of `math/base/special/pow`. Perhaps instead of randomly generating exponents in a loop, we can simply hardcode 3-4 examples (sign, frac, exp, etc) to show roundtripping. Does that sound reasonable, @gunjjoshi?

**Comment by gunjjoshi** [comment]
@kgryte Yes, that does. I'll update the examples accordingly, thanks!

**Comment by gunjjoshi** [comment]
Updated the C examples. Also, updated benchmarks to follow latest conventions.

**Comment by gunjjoshi** [comment]
The CI failure seems to occur due to the failure in `cephes` benchmarks, as it is unable to find `cephes` files.

**Comment by kgryte** [comment]
> The CI failure seems to occur due to the failure in cephes benchmarks

@Planeshifter I forget. Did we not already address the Cephes build issue?

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @gunjjoshi.

**Comment by Planeshifter** [comment]
@kgryte I don't recall us addressing the Cephes build errors.

**Comment by kgryte** [comment]
As the CI failure is due to an upstream workflow failure and not due to the changes in this PR, I will go ahead and merge.

**Pull request merged**